### PR TITLE
fix(backend): preserve DeepSeek V4 reasoning context

### LINKED ISSUE #257: feat: need deepseek v4 pro support / 加入对DS V4 PRO的支持
Currently, it seems to only support non-thinking LLMs and does not support Deekseek V4 PRO. Thinking LLMs contribute to the realization of iterative design capabilities.

当前似乎只支持非思考型模型，不支持DS V4 PRO，思考型模型应当有助于迭代设计能力的实现。

### PR DESCRIPTION
**Pull request by qinsz01**
## Summary
- Problem: DeepSeek V4 Pro thinking mode can return `reasoning_content`, and provider-compatible tool-call turns may reject follow-up requests when that field is dropped.
- Why it matters: StructureClaw routes agent turns through tool calls, so losing provider reasoning metadata can break DeepSeek V4 Pro conversations after a tool result.
- What changed: Preserve AI message metadata during graph message reconstruction and attach DeepSeek V4 `reasoning_content` back onto assistant messages before OpenAI-compatible chat completions requests.
- What did NOT change (scope boundary): Token-level LLM streaming remains disabled in the graph path; this PR does not restore streamed model tokens.

## Change Type (select all)
- [x] Bug fix
- [ ] Feature
- [x] Refactor required for the fix
- [ ] Docs
- [ ] Security hardening
- [ ] Chore/infra

## Scope (select all touched areas)
- [x] Gateway / orchestration
- [x] Skills / tool execution
- [ ] Auth / tokens
- [ ] Memory / storage
- [x] Integrations
- [x] API / contracts
- [ ] UI / DX
- [ ] CI/CD / infra

## Linked Issue/PR
- Closes #257
- [x] This PR fixes a bug or regression

## Root Cause / Regression History (if applicable)
- Root cause: LangChain preserves DeepSeek `reasoning_content` in `additional_kwargs`, but the OpenAI-compatible completions request conversion does not send that provider-specific field back on assistant messages.
- Missing detection / guardrail: There was no unit coverage for provider-specific assistant metadata surviving tool-call round trips.
- Prior context (`git blame`, prior PR, issue, or refactor if known): Issue #257 reported DeepSeek V4 Pro compatibility.
- Why this regressed now: DeepSeek V4 Pro thinking mode makes this field relevant for tool-call follow-up requests.
- If unknown, what was ruled out: N/A

## Regression Test Plan (if applicable)
- Coverage level that should have caught this:
- [x] Unit test
- [ ] Seam / integration test
- [ ] End-to-end test
- [ ] Existing coverage already sufficient
- Target test or file: `backend/src/utils/__tests__/llm-options.test.mjs`
- Scenario the test should lock in: Assistant messages with DeepSeek V4 `reasoning_content` keep that field when converted into chat completions request messages.
- Why this is the smallest reliable guardrail: It tests the exact compatibility shim without requiring a live DeepSeek API key.
- Existing test that already covers this (if any): N/A
- If no new test is added, why not: N/A

## User-visible / Behavior Changes
DeepSeek V4 Pro provider compatibility is improved for agent/tool-call conversations. No settings or UI changes.

## Diagram (if applicable)
```text
Before: DeepSeek V4 assistant tool call -> tool result -> follow-up request without reasoning_content -> provider may reject
After: DeepSeek V4 assistant tool call -> tool result -> follow-up request with reasoning_content -> provider-compatible continuation
```

## Security Impact (required)
- New permissions/capabilities? (`Yes/No`) No
- Secrets/tokens handling changed? (`Yes/No`) No
- New/changed network calls? (`Yes/No`) No
- Command/tool execution surface changed? (`Yes/No`) No
- Data access scope changed? (`Yes/No`) No
- If any `Yes`, explain risk + mitigation: N/A

### Environment
- OS: Windows
- Runtime/container: Node.js backend, local workspace
- Model/provider: DeepSeek V4 Pro compatibility path, tested without live provider call
- Integration/channel (if any): OpenAI-compatible chat completions
- Relevant config (redacted): N/A

### Steps
1. Build backend.
2. Run backend lint.
3. Run targeted Jest tests for LLM options and graph message state.

### Expected
- Backend compiles and lint passes.
- DeepSeek V4 reasoning metadata is attached to assistant chat completion messages.
- Existing graph state and empty response guards continue to pass.

### Actual
- All verification commands passed locally.

## Evidence
- [x] Failing test/log before + passing after
- [ ] Trace/log snippets
- [ ] Screenshot/recording
- [ ] Perf numbers (if relevant)

## Human Verification (required)
- Verified scenarios: `npm run build --prefix backend`; `npm run lint --prefix backend`; `npm test --prefix backend -- --runInBand src/utils/__tests__/llm-options.test.mjs src/agent-langgraph/__tests__/graph-state.test.mjs src/agent-langgraph/__tests__/graph-empty-response.test.mjs`
- Edge cases checked: Empty AI chunks without restorable metadata are still skipped; AI messages with tool calls or DeepSeek reasoning metadata are preserved.
- What you did **not** verify: Live DeepSeek V4 Pro API call with a real API key; restoring token-level LLM streaming.

## Review Conversations
- [x] I replied to or resolved every bot review conversation I addressed in this PR.
- [x] I left unresolved only the conversations that still need reviewer or maintainer judgment.

## Compatibility / Migration
- Backward compatible? (`Yes/No`) Yes
- Config/env changes? (`Yes/No`) No
- Migration needed? (`Yes/No`) No
- If yes, exact upgrade steps: N/A

## Risks and Mitigations
- Risk: The compatibility shim is scoped by model name and could miss differently named DeepSeek V4 variants.
- Mitigation: The matcher targets `deepseek-v4` case-insensitively, matching the reported model family while avoiding unrelated providers.

### CODE REVIEW

File: backend/src/utils/llm.ts [hunk 1]
```diff
@@ -33,6 +33,115 @@ export function buildChatModelOptions(
...
+    } finally {
+      this.activeSourceMessages = previous;
+    }
+  }
+
+  completionWithRetry(request: any, requestOptions?: any): Promise<any> {
+    const patchedRequest = attachDeepSeekReasoningContent(request, this.activeSourceMessages);
+    return super.completionWithRetry(patchedRequest as any, requestOptions);
+  }
+}
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/utils/llm.ts]: Fixed in d1d485f. I removed the shared `activeSourceMessages` field and now pass source messages via request-scoped options for both `_generate` and `_streamResponseChunks`, so concurrent calls cannot overwrite each other's source context.

File: backend/src/utils/llm.ts [hunk 2]
```diff
@@ -33,6 +33,115 @@ export function buildChatModelOptions(
...
+      yield* super._streamResponseChunks(messages, options, runManager);
+    } finally {
+      this.activeSourceMessages = previous;
+    }
+  }
+
+  completionWithRetry(request: any, requestOptions?: any): Promise<any> {
+    const patchedRequest = attachDeepSeekReasoningContent(request, this.activeSourceMessages);
+    return super.completionWithRetry(patchedRequest as any, requestOptions);
+  }
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/utils/llm.ts]: Fixed in d1d485f. The DeepSeek source messages are now carried through request-scoped call options instead of instance state, and the internal option is stripped before forwarding the request options to the OpenAI client.

File: backend/src/utils/llm.ts [hunk 3]
```diff
@@ -33,6 +33,115 @@ export function buildChatModelOptions(
...
+
+    const record = message as Record<string, unknown>;
+    if (hasOwn(record, 'reasoning_content')) return message;
+
+    changed = true;
+    return { ...record, reasoning_content: reasoningContent };
+  });
+
+  return changed ? { ...request, messages } : request;
+}
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/utils/llm.ts:110]: Fixed in d1d485f. I added a guard that bails out without attaching `reasoning_content` when the assistant counts diverge, so we avoid silently attaching reasoning to the wrong assistant message. Added a unit test for that mismatch case as the smallest reliable guardrail.

**Review by guyi2000 — APPROVED** [review]
